pkg/utils/utils.go

```diff
@@ -0,0 +1,43 @@
+package utils
+
+import (
+	openlibertyv1beta1 "github.com/OpenLiberty/open-liberty-operator/pkg/apis/openliberty/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Utility methods specific to Liberty and it's configuration
+
+// CustomizeEnv adds configured env variables appending configured liberty settings
+func CustomizeEnv(pts *corev1.PodTemplateSpec, la *openlibertyv1beta1.LibertyApplication) {
+	targetEnv := la.GetEnv()
+
+	// update or set to default depending on CRD config
+	if la.Spec.Logs != nil && la.Spec.Logs.ConsoleFormat != nil {
+		logVar := corev1.EnvVar{Name: "WLP_LOGGING_CONSOLE_FORMAT", Value: *la.Spec.Logs.ConsoleFormat}
+		targetEnv = append(targetEnv, logVar)
+	} else {
+		logVar := corev1.EnvVar{Name: "WLP_LOGGING_CONSOLE_FORMAT", Value: "json"}
+		if i, found := findEnvVar(logVar, pts.Spec.Containers[0].Env); found {
+			pts.Spec.Containers[0].Env[i] = logVar
+		} else {
+			targetEnv = append(targetEnv, logVar)
+		}
+	}
+
+	// validate if it's a new var before appending it to the pod spec
+	for _, v := range targetEnv {
+		if _, found := findEnvVar(v, pts.Spec.Containers[0].Env); !found {
+			pts.Spec.Containers[0].Env = append(pts.Spec.Containers[0].Env, v)
+		}
+	}
+}
+
+// findEnvVars checks if the environment variable is already present
+func findEnvVar(e corev1.EnvVar, envList []corev1.EnvVar) (int, bool) {
```

Comment:
I think its would be better to return &envList[i] (pointer) if found or nil otherwise

Reply:
Made the change in recent commit, along with the additional offline suggestions